lib/engine/game/g_1822/step/discard_train.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/discard_train'
+
+module Engine
+  module Game
+    module G1822
+      module Step
+        class DiscardTrain < Engine::Step::DiscardTrain
+          def process_discard_train(action)
```

Comment:
I'm a bit confused about why this is necessary

Reply:
In base DiscardTrain there is no check to see if the player discarding a train on a corporation is accually the owner. I got it when 2 corporations had to discard trains. Player 1 could discard a train in corporation 2 which had player 2 as an owner.

Didnt want to change the base code if this was intentional.

Reply:
Odd, it should've been player 2's turn at that point.

Reply:
When you comes to DiscardStep all corporations with an excess of train will show. Even if its player 1's turn both corporations with excess train will show.

Reply:
I just added the check so when a player click on a train in the step he/she have to be the owner of that specific corporation

Reply:
Can we separate the Discard train from this PR, I'd like to understand it more as it should only show one players trains.

Reply:
Sure

Reply:
The code for selecting which corps is crowded, this from base.

```ruby
      def crowded_corps
        @crowded_corps ||= corporations.select do |c|
          trains = self.class::OBSOLETE_TRAINS_COUNT_FOR_LIMIT ? c.trains.size : c.trains.count { |t| !t.obsolete }
          trains > train_limit(c)
        end
      end
```
There is nothing that stats it should be just for one entity

Reply:
And in DiscardTrain

```ruby
      def active_entities
        crowded_corps.take(1)
      end
```

Reply:
Yeah, other games immediately interrupt one player at a time to discard trains. So one corp does it then next and so on.

Is 1822 rules different?

Reply:
ahh, this should be a fix to the base class then and tested appropriately (and should be a different PR)

Reply:
Yep, i removed my overriden class.